File: Программирование/Лабораторные работы/Lab_1/Код/lab1.pas
program lab_1;
uses math;
const
start = -9;
endt = 2;
step = 0.3;
var x, y : real;
begin
  	x := start;
  	while (x <= endt) do
  	begin
    		if (x < -7) then
      		y := tan(x) / 71

    		else if (x >= -7) and (x < 0) then begin
      		x := round(x * 100) / 100;
      		y := ((x * x) / (exp(0.1 * (-x) * ln(-x))) * (log2(-x) / (x * x * x)))
    		end

    		else if x >= 0.3 then
      		y := (exp(0.1 * x * ln(x)) / cos(x) + x);

     		writeln('x = ', x:2:2, ' y = ', y:6:2);

    		x := x + step;
  	end;
readln;
end.

File: Программирование/Лабораторные работы/Lab_1/Код/lab1.cs
using System;

namespace lab1
{
    class math
    {
        public const double start = -9.00;
        public const double endt = 2.00;
        public const double step = 0.3;
    }
    class Program
    {
        static void Main(string[] args)
        {
            double y = 0;
            double x = math.start;
            while (x < math.endt)
            {
                if (x < -7)
                {
                    y = Math.Round(Math.Tan(x)/71,2);
                    
                }
                else
                {
                    if ((x >= -7) & (x < 0))
                    {
                        x = Math.Round((double)((x * 100) / 100),2);
                        y = Math.Round(((x * x) / (Math.Exp(0.1 * (-x) * Math.Log(-x)))) * ((Math.Log(2,(-x)) / (x * x * x))),2);
                        //  //Math.Pow(x,0.1*x)
                    }
                    else
                    {
                        if (x >= 0.3)
                        {
                            y = Math.Round((Math.Exp(0.1 * x * Math.Log(x)) / Math.Cos(x) + x),2);
                        }                        
                    }
                }
                Console.WriteLine("x = " +  x + " y = " + y);
                x = x + math.step;
                x = Math.Round((double)((x * 100) / 100), 2);
            }            
        }
    }
}